README.md

```diff
@@ -120,6 +127,65 @@ By default (when `emitCss: true`) the CSS styles will be emitted into a virtual
 
 If you set `emitCss: false` and your Svelte components contain `<style>` tags, the compiler will add JavaScript that injects those styles into the page when the component is rendered. That's not the default, because it adds weight to your JavaScript, prevents styles from being fetched in parallel with your code, and can even cause CSP violations.
 
+
+## Hot module replacement
+
+This plugin currently includes HMR support via community supported [svelte-hmr](https://github.com/rixo/svelte-hmr). This will be replaced by the official implementation once it is supported by Svelte's compiler itself.
+
+What this option does is add `esm-hmr` compatible HMR handlers (see [there](https://github.com/rixo/svelte-hmr#whats-hmr-by-the-way) for an in-depth explanation) to the compiled components code.
+
+By itself, this won't give you HMR out of the box with Rollup, because Rollup doesn't come with a dev server. See bellow for some HMR solutions with Rollup, or usage in HMR-enabled Vite.
+
+Note that the `dev` compiler option is required for HMR. **If `compilerOptions.dev` is not set or falsy, the `hot` option will be disabled too.** (Which is probably what you want for a production build.)
+
+### Options
+
+For now (while using the community implementation), you can pass any of [svelte-hmr's options](https://github.com/rixo/svelte-hmr#options) to the `hot` option of this plugin.
+
+For example:
+
+```js
+svelte({
+  compilerOptions: {
+    dev: true,
+  },
+  hot: {
+    preserveLocalState: true,
+  }
+})
+```
+
+### HMR with Rollup
+
+Rollup doesn't support HMR natively, because it doesn't even come with a dev server. If you want to enable HMR in a Rollup project, you'll need a HMR plugin like [rollup-plugin-hot](https://github.com/rixo/rollup-plugin-hot).
+
+Or you can use [Nollup](https://github.com/PepsRyuu/nollup), a Rollup compatible dev server intended for faster rebuild during development (note that you'd also need a `esm-hmr` compatibility plugin with Nollup, like [this](https://github.com/rixo/rollup-plugin-hot-nollup)).
```

Comment:
I'm not sure how much you want to support this option, but if we were to link to `rollup-plugin-hot-nollup` it'd probably be nice for that project to have a minimal README with some basic instructions